torch/ao/quantization/quantize.py

```diff
@@ -214,12 +214,12 @@ def insert_activation_post_process(m, special_act_post_process=None):
             # activation_post_process are now added directly to nn.Sequentail/_FusedModule
             if needs_observation(child):
                 insert_activation_post_process(child)
-        elif _has_special_act_post_process(child):
-            special_act_post_process = _get_special_act_post_process(child)
-            insert_activation_post_process(child, special_act_post_process)
         elif non_leaf_module_list is not None and type_before_parametrizations(child) in non_leaf_module_list:
             if needs_observation(child):
                 insert_activation_post_process(child)
+        elif _has_special_act_post_process(child):
```

Comment:
I think so. There's no validation logic for eager mode right now so we just override it here. Yes I think simply removing it will make things cleaner, though it'll be a broader scope change since it'll break a lot of existing use cases. I'll leave that for a future PR to unblock https://github.com/pytorch/pytorch/pull/88456.

Reply:
Filed https://github.com/pytorch/pytorch/issues/88579